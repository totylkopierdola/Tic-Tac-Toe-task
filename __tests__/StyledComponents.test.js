import React from "react";
import renderer from "react-test-renderer";
import {
  Container,
  Title,
  PlayerTurn,
  BoardWrapper,
  Board,
  Tile,
  ResetButton,
  Overlay,
  GameOverWrapper,
} from "../src/components/StyledComponents";

describe("Styled Components Snapshot Testing", () => {
  test("renders Container correctly", () => {
    const tree = renderer.create(<Container />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders Title correctly", () => {
    const tree = renderer.create(<Title>Game Title</Title>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders PlayerTurn correctly for player X", () => {
    const tree = renderer
      .create(<PlayerTurn $player="x">Player X's Turn</PlayerTurn>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders PlayerTurn correctly for player O", () => {
    const tree = renderer
      .create(<PlayerTurn $player="o">Player O's Turn</PlayerTurn>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders BoardWrapper correctly", () => {
    const tree = renderer.create(<BoardWrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders Board correctly", () => {
    const tree = renderer.create(<Board />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders Tile correctly when empty", () => {
    const tree = renderer
      .create(<Tile $value={null} $isClickable={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders Tile correctly for X", () => {
    const tree = renderer
      .create(<Tile $value="x" $isClickable={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders Tile correctly for O", () => {
    const tree = renderer
      .create(<Tile $value="o" $isClickable={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders ResetButton correctly", () => {
    const tree = renderer
      .create(<ResetButton>Reset Game</ResetButton>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders Overlay correctly", () => {
    const tree = renderer.create(<Overlay $show={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders GameOverWrapper correctly with a winner", () => {
    const tree = renderer
      .create(
        <GameOverWrapper $winner="x">
          <h2>Player X Wins!</h2>
        </GameOverWrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders GameOverWrapper correctly without a winner (draw)", () => {
    const tree = renderer
      .create(
        <GameOverWrapper $winner={null}>
          <h2>Draw!</h2>
        </GameOverWrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
