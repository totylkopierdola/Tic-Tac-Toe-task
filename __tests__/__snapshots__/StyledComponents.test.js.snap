// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Components Snapshot Testing renders Board correctly 1`] = `
<div
  className="sc-dsLQwm ilhPqq"
/>
`;

exports[`Styled Components Snapshot Testing renders BoardWrapper correctly 1`] = `
<div
  className="sc-fQpRED boxwnL"
/>
`;

exports[`Styled Components Snapshot Testing renders Container correctly 1`] = `
<div
  className="sc-braxZu fMttEx"
/>
`;

exports[`Styled Components Snapshot Testing renders GameOverWrapper correctly with a winner 1`] = `
<div
  className="sc-iCZwEW Ocnki"
>
  <h2>
    Player X Wins!
  </h2>
</div>
`;

exports[`Styled Components Snapshot Testing renders GameOverWrapper correctly without a winner (draw) 1`] = `
<div
  className="sc-iCZwEW cfuKOM"
>
  <h2>
    Draw!
  </h2>
</div>
`;

exports[`Styled Components Snapshot Testing renders Overlay correctly 1`] = `
<div
  className="sc-la-DxNn beEeuC"
/>
`;

exports[`Styled Components Snapshot Testing renders PlayerTurn correctly for player O 1`] = `
<div
  className="sc-elDIKY jFnifB"
>
  Player O's Turn
</div>
`;

exports[`Styled Components Snapshot Testing renders PlayerTurn correctly for player X 1`] = `
<div
  className="sc-elDIKY gIDxat"
>
  Player X's Turn
</div>
`;

exports[`Styled Components Snapshot Testing renders ResetButton correctly 1`] = `
<button
  className="sc-gnpbhQ hucphQ"
>
  Reset Game
</button>
`;

exports[`Styled Components Snapshot Testing renders Tile correctly for O 1`] = `
<div
  className="sc-iKTcqh ieIZJe"
/>
`;

exports[`Styled Components Snapshot Testing renders Tile correctly for X 1`] = `
<div
  className="sc-iKTcqh bcngpo"
/>
`;

exports[`Styled Components Snapshot Testing renders Tile correctly when empty 1`] = `
<div
  className="sc-iKTcqh bxBMRo"
/>
`;

exports[`Styled Components Snapshot Testing renders Title correctly 1`] = `
<h1
  className="sc-gJhJTp kPzxOS"
>
  Game Title
</h1>
`;
